detection_rules/exception.py

```diff
@@ -139,9 +142,37 @@ class Data(MarshmallowDataclassMixin):
 @dataclass(frozen=True)
 class TOMLExceptionContents(MarshmallowDataclassMixin):
     """Data stored in an exception file."""
+
     metadata: ExceptionMeta
     exceptions: List[Data]
 
+    @classmethod
+    def from_exceptions_dict(
+        cls,
+        exceptions_dict: dict,
+    ) -> "TOMLExceptionContents":
+        """Create a TOMLExceptionContents from a kibana rule resource."""
+        # Parse rule name and rule id from Kibana generated exception list name and description
+        rule_name = exceptions_dict["container"]["name"].replace("Exceptions for rule - ", "", 1)
```

Comment:
will this always be one rule?

Reply:
For standard rule anchored exceptions list yes. However, for shared exceptions list it the description is user provided and not auto generated. As such we will need to use a different method for finding the rule the container is linked to. Updating to address this now.